src/dart.rs

```diff
@@ -49,6 +54,91 @@ impl zed::Extension for DartExtension {
         Self
     }
 
+    /// ref:
+    /// https://github.com/zed-industries/zed/blob/main/crates/dap_adapters/src/gdb.rs
+    fn get_dap_binary(
+        &mut self,
+        adapter_name: String,
+        config: DebugTaskDefinition,
+        user_provided_debug_adapter_path: Option<String>,
+        worktree: &Worktree,
+    ) -> Result<DebugAdapterBinary, String> {
+        let user_config: serde_json::Value = serde_json::from_str(&config.config)
+            .map_err(|e| format!("Failed to parse debug config: {}", e))?;
+
+        let program = user_config
+            .get("program")
+            .and_then(|v| v.as_str())
+            .unwrap_or("lib/main.dart");
+
+        let args = user_config
+            .get("args")
+            .and_then(|v| v.as_array())
+            .map(|arr| {
+                arr.iter()
+                    .filter_map(|v| v.as_str())
+                    .map(|s| s.to_string())
+                    .collect::<Vec<String>>()
+            })
+            .unwrap_or_else(|| vec![]);
+
+        // Get debug_mode from user config (flutter or dart)
+        let debug_mode = user_config
+            .get("type")
+            .and_then(|v| v.as_str())
+            .filter(|s| !s.trim().is_empty()) // Filter out empty strings
+            .ok_or_else(|| "type is required and cannot be empty or null".to_string())?;
+        let command = if debug_mode == "flutter" {
+            "flutter"
+        } else {
+            "dart"
+        };
+        let device_id = user_config
+            .get("device_id")
+            .and_then(|v| v.as_str())
+            .unwrap_or("chrome");
+
+        let platform = user_config
+            .get("platform")
+            .and_then(|v| v.as_str())
+            .unwrap_or("web");
+        let cwd = Some(worktree.root_path());
+
+        let config_json = json!({
+            "type": command,
+            "request": "launch",
+            "program": program,
+            "cwd": cwd.clone().unwrap_or_default(),
+            "args": args,
+            "flutterMode": "debug",
+            "deviceId": device_id,
+            "platform": platform,
+            "stopOnEntry": false
+        })
+        .to_string();
+
+        let debug_adapter_binary = DebugAdapterBinary {
+            command: Some(command.to_string()),
+            arguments: vec!["debug_adapter".to_string()],
+            envs: vec![], // Add any Dart-specific env vars if needed
```

Comment:
```suggestion
            envs: env, // Add any Dart-specific env vars if needed
```

Could be nice to pass `env` param here. I didn't quite get it working, but I was trying something like this:

```rust
let env = match user_config.get("env").and_then(|v| v.as_object()) {
            Some(env_vars) => env_vars
                .iter()
                .map(|(k, v)| (k.clone(), v.to_string()))
                .collect(),
            None => vec![],
        };
```

Thanks for the great work on this!